source/postprocess/visualization/boundary_indicator.cc

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/boundary_indicator.h>
+#include <aspect/simulator_access.h>
+
+
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      std::pair<std::string, Vector<float> *>
+      BoundaryIndicator<dim>::execute() const
+      {
+        std::pair<std::string, Vector<float> *>
+        return_value ("boundary_indicator",
+                      new Vector<float>(this->get_triangulation().n_active_cells()));
+
+        // loop over all of the surface cells and extract boundary indicators
+        typename DoFHandler<dim>::active_cell_iterator
+        cell = this->get_dof_handler().begin_active(),
+        endc = this->get_dof_handler().end();
+
+        unsigned int cell_index = 0;
+        for (; cell!=endc; ++cell,++cell_index)
+          if (cell->is_locally_owned())
+            if (cell->at_boundary())
+              {
+                types::boundary_id boundary_id;
+                for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+                  {
+                    boundary_id = cell->face(f)->boundary_indicator();
+                    if (boundary_id != numbers::internal_face_boundary_id)
+                      break;
+                  }
+                (*return_value.second)(cell_index) = static_cast<unsigned int> (boundary_id);
+              }
+        // internal cells are all set to the same boundary indicator value 2*dim+2*(dim-1).
+        // when using the TwoMergedBoxes geometry model, this is the value following upon
+        // the last used boundary indicator value
```

Comment:
I think that this visualization postprocessor is actually quite useful for any kind of geometry, not just the two cubes geometry. So I see no reason to set internal cells to this particular value.

If anything, I would say ask the geometry for the set of used boundary indicators and set internal cells to the largest of these values plus one. I think this should work via
```
  const types::boundary_id largest_boundary_id_plus_one = *this->get_geometry_model().get_used_boundary_indicators().rbegin() + 1;
```
at the top of the function.